test/TestUtil.js

```diff
@@ -30,6 +29,30 @@ exports.promiseLater = function (time, func) {
     }));
 };
 
+/**
+ * Returns rejection reason if rejected, otherwise returns a dummy error.
+ */
+exports.getRejectionReasonOrThrow = async function (asyncFn) {
+    try {
+        await asyncFn();
+        throw new Error('Expected the call the throw, but it didn\'t.');
+    } catch (e) {
+        return e;
+    }
+};
+
+/**
+ * Returns thrown error if thrown, otherwise returns a dummy error.
+ */
+exports.getThrownErrorOrDummy = function (fn) {
```

Comment:
We should probably rename this to `getThrownErrorOrThrow`. Also, it would be good to update documentation of these two functions

Reply:
I think `returns a dummy error` is accurate. What do you prefer?

Reply:
Oh, I missed it. I think it would be much better if we change it to throw. This way, if `fn()` does not throw, the test will automatically fail with the error message below. Otherwise, the test may again fail but with a different error message such as expected a different error type, etc. which is not really intuitive.

Reply:
makes sense, same for getRejectionReason right?

Reply:
yes